From 3c16612eed3fbeaf3c4d2876dc27f98f46fcf0d8 Mon Sep 17 00:00:00 2001
From: System233 <cdkeylzh@outlook.com>
Date: Sun, 14 Jun 2026 19:46:00 +0800
Subject: [PATCH] Fix MSVC library naming for static builds

---
 configure | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/configure b/configure
index f164e15..761c0d7 100644
--- a/configure
+++ b/configure
@@ -3998,6 +3998,18 @@ msvc_common_flags(){
             -march=*)             ;;
-            -lz)                  echo zlib.lib ;;
+            -lz)                  echo zs.lib ;;
             -lx264)               echo libx264.lib ;;
+            -lx265)               echo libx265.lib ;;
+            -lxml2)               echo libxml2.lib ;;
+            -lwebp)               echo libwebp.lib ;;
+            -lmfx)                echo libmfx.lib ;;
+            -lmp3lame)            echo libmp3lame.lib ;;
+            -lopengl32)           echo opengl32.lib ;;
+            -lssl)                echo libssl.lib ;;
+            -lcrypto)             echo libcrypto.lib ;;
+            -ltesseract)          echo tesseract55.lib ;;
+            -lcurl)               echo libcurl.lib ;;
+            -lpng)                echo libpng16.lib ;;
+            -lm)                  ;;
             -lstdc++)             ;;
             -l*)                  echo ${flag#-l}.lib ;;
             -LARGEADDRESSAWARE)   echo $flag ;;
-- 
2.47.1.windows.2

