diff --git a/configure b/configure
index f91d2c2..c2ddab0 100644
--- a/configure
+++ b/configure
@@ -3944,8 +3944,15 @@ armasm_flags(){
     for flag; do
         case $flag in
             # Filter out MSVC cl.exe options from cflags that shouldn't
-            # be passed to gas-preprocessor
+            # be passed to gas-preprocessor or armasm64
             -M[TD]*)                                            ;;
+            -D*)                                                ;;
+            -U*)                                                ;;
+            -f*)                                                ;;
+            -m*)                                                ;;
+            -O*)                                                ;;
+            -W*)                                                ;;
+            -std*)                                              ;;
             *)                  echo $flag                      ;;
         esac
    done
