diff --git a/configure b/configure
index 9d6457d..938f243 100644
--- a/configure
+++ b/configure
@@ -6562,8 +6562,14 @@ enabled libgsm            && { for gsm_hdr in "gsm.h" "gsm/gsm.h"; do
                                    check_lib libgsm "${gsm_hdr}" gsm_create -lgsm && break;
                                done || die "ERROR: libgsm not found"; }
 enabled libilbc           && require libilbc ilbc.h WebRtcIlbcfix_InitDecode -lilbc $pthreads_extralibs
-enabled libjxl            && require_pkg_config libjxl "libjxl >= 0.7.0" jxl/decode.h JxlDecoderVersion &&
-                             require_pkg_config libjxl_threads "libjxl_threads >= 0.7.0" jxl/thread_parallel_runner.h JxlThreadParallelRunner
+enabled libjxl            && { check_pkg_config libjxl "libjxl >= 0.7.0" jxl/decode.h JxlDecoderVersion ||
+                                { { check_lib libjxl jxl/decode.h JxlDecoderVersion "-ljxl -lhwy -lbrotlienc -lbrotlidec -lbrotlicommon -ljxl_cms -llcms2" ||
+                                require libjxl jxl/decode.h JxlDecoderVersion "-ljxl -lhwy -lbrotlienc -lbrotlidec -lbrotlicommon -ljxl_cms -llcms2"; } &&
+                                    warn "using libjxl without pkg-config"; } } &&
+                             { check_pkg_config libjxl_threads "libjxl_threads >= 0.7.0" jxl/thread_parallel_runner.h JxlThreadParallelRunner ||
+                                { { check_lib libjxl_threads jxl/thread_parallel_runner.h JxlThreadParallelRunner "-ljxl_threads" ||
+                                require libjxl_threads jxl/thread_parallel_runner.h JxlThreadParallelRunner "-ljxl_threads"; } &&
+                                    warn "using libjxl_threads without pkg-config"; } }
 enabled libklvanc         && require libklvanc libklvanc/vanc.h klvanc_context_create -lklvanc
 enabled libkvazaar        && require_pkg_config libkvazaar "kvazaar >= 0.8.1" kvazaar.h kvz_api_get
 enabled liblensfun        && require_pkg_config liblensfun lensfun lensfun.h lf_db_new
