diff --git a/ffbuild/common.mak b/ffbuild/common.mak
index 0f5acd7..75f74c9 100644
--- a/ffbuild/common.mak
+++ b/ffbuild/common.mak
@@ -122,7 +122,7 @@ COMPILE_LASX = $(call COMPILE,CC,LASXFLAGS)
 	$(Q)echo '#include "$*.h"' >$@
 
 RUN_BIN2C = $(BIN2C) $(patsubst $(SRC_PATH)/%,$(SRC_LINK)/%,$<) $@ $(subst .,_,$(basename $(notdir $@)))
-RUN_GZIP  = $(M)gzip -nc9 $(patsubst $(SRC_PATH)/%,$(SRC_LINK)/%,$<) >$@
+RUN_GZIP  = $(M)mkdir -p $(@D) && gzip -nc9 $(patsubst $(SRC_PATH)/%,$(SRC_LINK)/%,$<) >$@
 RUN_MINIFY = $(M)sed 's!/\\*.*\\*/!!g' $< | tr '\n' ' ' | tr -s ' ' | sed 's/^ //; s/ $$//' > $@
 %.gz: TAG = GZIP
 %.min: TAG = MINIFY
