diff --git a/libswscale/Makefile b/libswscale/Makefile
index 5e5c7c45fc..86dfa6b634 100644
--- a/libswscale/Makefile
+++ b/libswscale/Makefile
@@ -56,8 +56,6 @@ OBJS-$(CONFIG_UNSTABLE) +=                              \
        uops_backend.o                                  \
 
-ifeq ($(CONFIG_UNSTABLE),yes)
 include $(SRC_PATH)/libswscale/vulkan/Makefile
-endif
 
 # Objects duplicated from other libraries for shared builds
 SHLIBOBJS                    += log2_tab.o half2float.o
