From 61c6841bb13d8e711cf15b050be917533a6a676c Mon Sep 17 00:00:00 2001
From: System233 <cdkeylzh@outlook.com>
Date: Mon, 15 Jun 2026 11:34:28 +0800
Subject: [PATCH] Fix iconv static linking on MSVC

---
 configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure b/configure
index 8ff1828..dae8d0c 100644
--- a/configure
+++ b/configure
@@ -6848,7 +6848,7 @@ enabled amf &&
 
 # Funny iconv installations are not unusual, so check it after all flags have been set
 if enabled libc_iconv; then
-    check_func_headers iconv.h iconv
+    check_lib iconv iconv.h iconv -liconv || check_lib iconv iconv.h iconv -liconv -llibcharset
 elif enabled iconv; then
     check_func_headers iconv.h iconv || check_lib iconv iconv.h iconv -liconv || check_lib iconv iconv.h iconv -liconv -llibcharset
 fi
-- 
2.47.1.windows.2

